src/bin/gosub-parser.rs

```diff
@@ -1,28 +1,28 @@
-use anyhow::Result;
+use anyhow::bail;
 use gosub_html5::parser::document::{Document, DocumentBuilder};
 use gosub_html5::parser::Html5Parser;
 use gosub_shared::bytes::{CharIterator, Confidence, Encoding};
+use gosub_shared::types::Result;
 use std::fs;
-use std::process::exit;
-
-fn bail(message: &str) -> ! {
-    println!("{message}");
-    exit(1);
-}
 
 fn main() -> Result<()> {
-    let url = std::env::args()
-        .nth(1)
-        .unwrap_or_else(|| bail("Usage: gosub-parser <url>"));
+    let matches = clap::Command::new("Gosub parser")
+        .version("0.1.0")
+        .arg(
+            clap::Arg::new("url")
+                .help("The url or file to parse")
+                .required(true)
+                .index(1),
+        )
+        .get_matches();
+
+    let url: String = matches.get_one::<String>("url").expect("url").to_string();
```

Comment:
I'd probably use `.ok_or("missing url")?` here instead of `.expect` since it fits the `-> Result<()>` main function better.

Reply:
```
    let url: String = matches
        .get_one::<String>("url")
        .ok_or("Missing url")
        .unwrap()
        .to_string();
```